### DISCUSSION TITLE
How to format single citation reference and references list?

### DISCUSSION THREAD

**Discussion by aborruso** [quarto-dev/quarto-cli#8033] [Q&A] [answered]
### Description

Hi,
I'm building a quarto book project. [Code](url) and [output](https://aborruso.github.io/linee-guida-opendata/).

I have some goals related to my [bibliography file](https://github.com/aborruso/linee-guida-opendata/blob/main/references.bib).

The first one is to cite one of its items and in the output to have its label and the link to it.

To write something like `See at [@CAD]`, and to have back "See at [CAD](https://www.normattiva.it/atto/caricaDettaglioAtto?atto.dataPubblicazioneGazzetta=2005-05-16&atto.codiceRedazionale=005G0104&atto.articolo.numero=0&atto.articolo.sottoArticolo=1&atto.articolo.sottoArticolo1=10&qId=b8226c0a-4e5d-4b0f-814e-ed010e5ef111&tabID=0.755394932407083&title=lbl.dettaglioAtto)".

In the [default way](https://aborruso.github.io/linee-guida-opendata/gruppo-di-lavoro.html) I have "See at ([«"Decreto legislativo 7 marzo 2005, n. 82 recante ""Codice dell’amministrazione digitale"""», s.d.](https://aborruso.github.io/linee-guida-opendata/references.html#ref-CAD))".

Reading the [documentation](https://quarto.org/docs/authoring/footnotes-and-citations.html#sec-citations), it does not seem possible to do so.

In the [reference page](https://aborruso.github.io/linee-guida-opendata/references.html), I have the default style. I would like to have back different template, something like:

- `**label**, [title](url)`

In my example:

- **CAD**, [Decreto Legislativo 7 Marzo 2005, n. 82 Recante ""Codice Dell’amministrazione Digitale" ](https://www.normattiva.it/atto/caricaDettaglioAtto?atto.dataPubblicazioneGazzetta=2005-05-16&atto.codiceRedazionale=005G0104&atto.articolo.numero=0&atto.articolo.sottoArticolo=1&atto.articolo.sottoArticolo1=10&qId=b8226c0a-4e5d-4b0f-814e-ed010e5ef111&tabID=0.755394932407083&title=lbl.dettaglioAtto)

And a last related question. I have in the [output](https://aborruso.github.io/linee-guida-opendata/gruppo-di-lavoro.html) `s.d.`, I think because I have no "date" in the reference file. Is there any way not to have it in the output?

![image](https://github.com/quarto-dev/quarto-cli/assets/30607/b313e2bb-f26e-4ce0-b53e-8e9e198df69e)

Thank you and best wishes 🎄

**Comment by aborruso** [comment]
Hi,
I have update my site and partially solved my questions.

I have created a [cls file](https://github.com/aborruso/linee-guida-opendata/blob/main/lg.csl) and now I can reference a reference id and have in output only the ID and the link to reference.
I write `See at [@CAD]` and I get

![image](https://github.com/quarto-dev/quarto-cli/assets/30607/41c34874-5558-449a-93c9-25028d6bcda8)

But I have two opened problems:
- in the [reference page](https://aborruso.github.io/linee-guida-opendata/references.html) I have only one item, and not all the [bib items](https://github.com/aborruso/linee-guida-opendata/blob/main/references.bib);
- In the cls I have set a [style](https://github.com/aborruso/linee-guida-opendata/blob/main/lg.csl#L10-L18) for citations list, But it is not applied. Why? I have [set](https://github.com/aborruso/linee-guida-opendata/blob/main/references.qmd#L3) `nocite: '@*'`

Thank you

**Reply by cderv** [reply]
By cloning your repo, I noticed that if I do `quarto render references.qmd` it will probuce all the references, but `quarto render` on the project does not. 🤔 

Can you confirm this ?

**Reply by cderv** [reply, marked as answer]
Could you set this in your `_quarto.yml` ? 
https://github.com/aborruso/linee-guida-opendata/blob/21376661731318600ab7e3e94884bd08da9820ae/references.qmd#L1-L5

 ```yaml
bibliography: references.bib
nocite: '@*'
csl: lg.csl
````

and not in the `references.qmd` document directly ?

**Reply by cderv** [reply]
It seems you did already... 🤔 
https://github.com/aborruso/linee-guida-opendata/blob/adc790d6eb71a8e01c6949838697dffbd6715781/_quarto.yml#L64-L66

I can't reproduce the issue on a demo book by setting these fields in `_quarto.yml` 

* I created a template book using `quarto create project`
* Added in `intro.qmd` the follow to generate a bib file 
````markdown
```{r}
#| include: false
knitr::write_bib(c("knitr", "rmarkdown"), "references.bib")
```
````
* Added no reference to any of those, but added this config in `_quarto.yml`
````yaml
bibliography: references.bib
nocite: '@*'
````

It works ok using `quarto render`

**Reply by aborruso** [reply]
> and not in the references.qmd document directly ?

Thank you @cderv, it works. But is it correct that it didn't work the way I had set it up?

Aggiungo un'altra domanda. I would like to format this list, and have for example in bold the id label.
I've made many attempts, but if I enter `**` or `<b>` [in my csl file](https://github.com/aborruso/linee-guida-opendata/blob/main/lg.csl) I don't get the bold, but the print of `**` and `<b>`.

Do you have an example to show me how I could do that?

Thank you again

**Reply by aborruso** [reply]
> It seems you did already... 🤔

Yes, 6 minutes ago, after your precious reply. I wanted to test before I answered you.

![image](https://github.com/quarto-dev/quarto-cli/assets/30607/17ecf18c-d435-4b8a-a69c-87f5126757da)

**Reply by cderv** [reply]
> I would like to format this list, and have for example in bold the id label.

This would be CSL syntax. I don't know how to tweak it. To write your own CSL, you need to look into CSL spec (I see a formatting section https://docs.citationstyles.org/en/stable/specification.html#formatting) and adapt it. 

You'll find some example in the Citation Style github repo: https://github.com/search?q=repo%3Acitation-style-language%2Fstyles%20font-weight&type=code

Hopefully you'll manage to do it

**Reply by aborruso** [reply]
Thank you again @cderv

**Reply by cderv** [reply]
> > It seems you did already... 🤔
> 
> Yes, 6 minutes ago, after your precious reply. I wanted to test before I answered you.
> 
> ![image](https://private-user-images.githubusercontent.com/30607/293044245-17ecf18c-d435-4b8a-a69c-87f5126757da.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM2ODYxMDMsIm5iZiI6MTcwMzY4NTgwMywicGF0aCI6Ii8zMDYwNy8yOTMwNDQyNDUtMTdlY2YxOGMtZDQzNS00YjhhLWE2OWMtODdmNTEyNjc1N2RhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI3VDE0MDMyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNjYzYzZDliNjUzMjk3Y2VlZDA0N2UwNzVhOGEzY2U0YTEyMjllZDAzMmRjNjkxNTUyZjk2ZjM1YTFlOWIzYmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ZVi8hM-3TxhMnw-h_cuF06s3OhdTKNVaq_HhZBSJIeI)

Indeed! 

By setting some YAML block into a document only, it will apply only when rendering this document. You cannot set some global config like `bibliography` this way because it should apply to all the documents (to resolve references). Citations is definitely a global feature for books
